dask/dataframe/io/csv.py

```diff
@@ -344,20 +351,19 @@ def read_pandas(reader, urlpath, blocksize=AUTO_BLOCKSIZE, collection=True,
     # Get header row, and check that sample is long enough. If the file
     # contains a header row, we need at least 2 nonempty rows + the number of
     # rows to skip.
-    skiprows = kwargs.get('skiprows', 0)
     names = kwargs.get('names', None)
     header = kwargs.get('header', 'infer' if names is None else None)
     need = 1 if header is None else 2
-    parts = b_sample.split(b_lineterminator, skiprows + need)
+    parts = b_sample.split(b_lineterminator, lastskiprow + need)
     # If the last partition is empty, don't count it
     nparts = 0 if not parts else len(parts) - int(not parts[-1])
 
-    if nparts < skiprows + need and len(b_sample) >= sample:
+    if nparts < lastskiprow + need and len(b_sample) >= sample:
```

Comment:
I assume this is tested, but adding a new test with `skip=[1, large_number]` would be good to verify I think.

Reply:
I wrote some tests and ended up feeling like the functionality was kind of broken for skip=large_number such that skip is not contained in the first partition. I tried to make it fail in a more useful way in this PR.

Reply:
let me know what you think.